pyrealm/core/solar.py

```diff
@@ -1,26 +1,584 @@
-"""The ``solar`` submodule provides functions and classes to calculate daily solar
-radiation fluxes and other radiative values.
+"""The :mod:`~pyrealm.core.solar` submodule provides functions to calculate
+photosynthetic photon flux density (ppfd), daily solar radiation fluxes and other
+radiative values.
 """  # noqa: D205
 
 import numpy as np
 from numpy.typing import NDArray
 
 from pyrealm.constants import CoreConst
+from pyrealm.core.utilities import check_input_shapes
 
-# from pyrealm.splash.const import (
-#     kA,
-#     kalb_sw,
-#     kalb_vis,
-#     kb,
-#     kc,
-#     kd,
-#     ke,
-#     keps,
-#     kfFEC,
-#     kGsc,
-#     komega,
-#     pir,
-# )
+
+def calc_distance_factor(nu: NDArray, k_e: float) -> NDArray:
+    r"""Calculates distance factor.
+
+    This function calculates distance factor using the method of Berger et al. (1993)
+
+    .. math::
+
+        dr = \left( \frac{1.0}{\left(\frac{1.0 - k_e^2}{1.0 + k_e \cos\left(\deg2rad(\nu)\right)}\right)} \right)^2
+
+    Args:
+        nu          : heliocentric true anomaly (degrees)
+        k_e         : Solar eccentricity
```

Comment:
```suggestion
    This function calculates the distance factor :math:`dr` using the method of
    :cite:t:`berger:1993a`.

    .. math::

        dr = \left( 1.0 \mathbin{/}
               \left(\frac{1.0 - k_e^2}
                          {1.0 + k_e \cos\left(\nu \cdot \pi \mathbin{/} 180)\right)}
               \right)
             \right)^2

    Args:
        nu          : heliocentric true anomaly (:math:`\nu`, degrees)
        k_e         : Solar eccentricity (:math:`k_e`)
```

This is exceptionally picky:
* I think the latex is easier to read with some line breaks but it also means we can drop the `noqa`. 
* I've also avoided the triple stack division - the symbols get tiny when it is rendered
* I've replaced `\deg2rad` with the actual conversion.
* I've annotated the text to identify the symbols in the equation.